### PR TITLE
Add PHP linting and compat actions

### LINKED ISSUE #78: Add linting & php compat actions
### Is your enhancement related to a problem? Please describe.

In order to ensure future PRs on this repo match our expectations we should add in our linting and PHP Compat GH actions (and any other default actions that aren't already in place here).

### Designs

See base set of actions here: https://github.com/10up/embed-block-figma/tree/develop/.github/workflows

### Describe alternatives you've considered

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by jjgrainger**
### Description of the Change
PR adds PHP linting and compat workflows.

Closes #78 

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added [Critical Flows, Test Cases, and/or End-to-End Tests](https://10up.github.io/Open-Source-Best-Practices/testing/) to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: phpcs-compat.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0"?>
+<ruleset name="PHPCompatibilityWP">
+	<description>PHP Compatibility checks only</description>
+
+	<rule ref="PHPCompatibilityWP" />
+	
+    <!-- Files to check -->
+	<arg name="extensions" value="php"/>
```
Comment (faisal-alvi) [inline phpcs-compat.xml]:
```suggestion
	<!-- Files to check -->
	<arg name="extensions" value="php"/>
```

File: .github/workflows/php-compat.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    branches:
+      - develop
+      - trunk
+    paths:
+      - "**.php"
+  pull_request:
+    branches:
+      - develop
+    paths:
+      - "**.php"
```
Comment (faisal-alvi) [inline .github/workflows/php-compat.yml:14]: Could you please add the PR branch `add/linting-php-compat-actions` to this list? This way, we can verify if the actions run and pass as expected.

**Review by faisal-alvi — CHANGES_REQUESTED** [review]
@jjgrainger Thanks for the PR! I’ve added a couple of suggestions below. Additionally, it would be great to add a blank line at the end of each file for consistency.

**Review by faisal-alvi — CHANGES_REQUESTED** [review]
Seems like it did not trigger new checks on a PR, we’ll need to update the branches list under `pull_request` like this:

```
pull_request:
    branches:
      - develop
      - add/linting-php-compat-actions
```

**Comment by iamdharmesh** [comment]
@faisal-alvi @jjgrainger The GitHub Actions workflow was not triggering because we have a PHP file paths filter in the workflow triggers, and there were no changes in any PHP files. I made a minor change in a PHP file to trigger the workflow, and it is working fine now. @faisal-alvi, could you please review this once again?

**Review by faisal-alvi — APPROVED** [review]

**Pull request merged**